### PR TITLE
Adding -AzureStorageAccounts parameter to Set-AzureRmWebApp and Set-AzureRmWebAppSlot

### PR DESCRIPTION
**Pull request by vinisoto**
## Description

Adding -AzureStorageAccounts parameter to Set-AzureRmWebApp and Set-AzureRmWebAppSlot to support "Bring your own storage" to Container-based web apps

## Checklist

- [x ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ x] If applicable, the changes made in the PR have proper test coverage
- [x ] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) [design review open] (https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/166): 
    - [x ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

## Description

Adding -AzureStorageAccounts parameter to Set-AzureRmWebApp and Set-AzureRmWebAppSlot to support "Bring your own storage" to Container-based web apps

## Checklist

- [x ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ x] If applicable, the changes made in the PR have proper test coverage
- [x ] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) [design review open] (https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/166): 
    - [x ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by vinisoto** [comment]
+ @jvano. Who else need to review this? Thanks.

**Comment by vinisoto** [comment]
Hi! I opened a design review for this changes: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/166

**Comment by vinisoto** [comment]
Hi @maddieclayton - I have submitted the changes as per the design review. However, looks like the [Jenkins build](https://azuresdkci.westus2.cloudapp.azure.com/job/powershell/8258/) is [failing](https://azuresdkci.westus2.cloudapp.azure.com/job/powershell/8258/artifact/msbuild.err):

`c:\workspace\powershell\src\packages\BuildTools.StyleCop.4.7.49.0\tools\StyleCop.targets(98,5): error MSB4062: The "StyleCopTask" task could not be loaded from the assembly c:\workspace\powershell\src\packages\BuildTools.StyleCop.4.7.49.0\tools\StyleCop.dll. Could not load file or assembly 'Microsoft.Build.Utilities.v3.5, Version=3.5.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [c:\workspace\powershell\src\ServiceManagement\RecoveryServices\Commands.RecoveryServices\Commands.RecoveryServicesRdfe.csproj]`

It doesn't seem related to my change but I might be wrong. Would you mind taking a look?

Thank you

File: src/ResourceManager/Websites/Commands.Websites/help/New-AzureRmWebAppAzureStoragePath.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
+---
+external help file: Microsoft.Azure.Commands.Websites.dll-Help.xml
+Module Name: AzureRM.Websites
+online version:
```
Comment (cormacpayne) [inline src/ResourceManager/Websites/Commands.Websites/help/New-AzureRmWebAppAzureStoragePath.md]: @vinisoto the build is currently failing because there's no value for this field. From the `ValidateHelpIssues.csv` file in the build artifacts:

_No entry was found for the online version field in the header. The corresponding URL should be the following: https://docs.microsoft.com/en-us/powershell/module/azurerm.websites/new-azurermwebappazurestoragepath_

Comment (vinisoto) [reply src/ResourceManager/Websites/Commands.Websites/help/New-AzureRmWebAppAzureStoragePath.md]: Thanks @cormacpayne - that solved it

**Comment by vinisoto** [comment]
Hey @maddieclayton and @cormacpayne looks like all checks have passed.

File: src/ResourceManager/Websites/Commands.Websites.Test/ScenarioTests/WebAppSlotTests.cs [hunk 2]
```diff
@@ -122,5 +122,12 @@ public void TestWebAppSwapWithPreviewCompleteSlotSwap()
         {
             WebsitesController.NewInstance.RunPsTest(_logger, "Test-WebAppSwapWithPreviewCompleteSlotSwap");
         }
+
+        [Fact(Skip = "Currently the API fails when adding Azure Storage Accounts for slots. Will enable this test when the API is fixed")]
```
Comment (maddieclayton) [inline src/ResourceManager/Websites/Commands.Websites.Test/ScenarioTests/WebAppSlotTests.cs]: We can't merge this PR until the API is fixed

File: src/ResourceManager/Websites/Commands.Websites/ChangeLog.md [hunk 3]
```diff
@@ -27,7 +28,6 @@
 * Updating to use the latest .NET SDK version (2.0.0)
 * Added two new cmdlets: Get-AzureRmDeletedWebApp and Restore-AzureRmDeletedWebApp
 * New-AzureRmAppServicePlan -HyperV switch is added for create app service plan with windows container
-* New-AzureRmWebApp/ New-AzureRmWebAppSlot/ Set-AzureRmWebApp/ Set-AzureRmWebAppSlot - New parameters (–ContainerRegistryUser string -ContainerRegistryPassword secureString -EnableContainerContinuousDeployment) added for creating and managing windows container app
```
Comment (maddieclayton) [inline src/ResourceManager/Websites/Commands.Websites/ChangeLog.md]: Please add this line back in.

File: src/ResourceManager/Websites/Commands.Websites/Validations/ValidateAzureStorageAccountsAttribute.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Management.WebSites.Models;
+using System;
+using System.Collections;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.WebApps.Validations
+{
+    public class ValidateAzureStorageAccountsAttribute : ValidateArgumentsAttribute
```
Comment (maddieclayton) [inline src/ResourceManager/Websites/Commands.Websites/Validations/ValidateAzureStorageAccountsAttribute.cs]: do you still need this class?

File: tools/AzureRM/AzureRM.psm1 [hunk 5]
```diff
@@ -1 +0,0 @@
-#placeholder
```
Comment (maddieclayton) [inline tools/AzureRM/AzureRM.psm1]: Please revert this change.

File: src/ResourceManager/Websites/Commands.Websites.Test/ScenarioTests/WebAppSlotTests.cs (hunk 2 above)
Comment (vinisoto) [reply src/ResourceManager/Websites/Commands.Websites.Test/ScenarioTests/WebAppSlotTests.cs]: @maddieclayton - I will ping you once the API is fixed

Comment (markcowl) [reply src/ResourceManager/Websites/Commands.Websites.Test/ScenarioTests/WebAppSlotTests.cs]: @vinosoto still waiting for this change

**Comment by markcowl** [comment]
@vinosoto  We are still waiting on re-enabling the test.  If this is not done by EOW, we will have to close this PR for lack of activity
